src/main.ts

```diff
@@ -20,9 +20,15 @@ import { platformBrowserDynamic } from "@angular/platform-browser-dynamic";
 
 import { AppModule } from "./app/app.module";
 import { environment } from "./environments/environment";
+import { AppConfig } from "./app/core/app-config/app-config";
 
 if (environment.production) {
   enableProdMode();
 }
 
-platformBrowserDynamic().bootstrapModule(AppModule);
+/**
+ * Loading AppConfig before bootstrap process (see {@link https://stackoverflow.com/a/66957293/10713841})
+ */
+AppConfig.load().then(() =>
+  platformBrowserDynamic().bootstrapModule(AppModule)
+);
```

Comment:
```suggestion
await AppConfig.load();
platformBrowserDynamic().bootstrapModule(AppModule);
```

can't we use a top level `await` here? :blush:

Reply:
No, await can only be used inside (`async`) functions. We could move the whole block into a async function and then call this function.

```
async function startApplication() {
  await AppConfig.load();
  platformBrowserDynamic().bootstrapModule(AppModule);
}

startApplication();
```

Reply:
https://typescript.tv/new-features/top-level-await-in-typescript-3-8/